Improve version control consistentency (#39616)
mdn/content: files/en-us/learn_web_development/core/version_control/index.md

{{PreviousMenu("Learn_web_development/Core/Design_for_developers", "Learn_web_development/Core")}}

> Version control tools (often called **Version Control Systems**, or **VCS**) are an essential part of modern workflows, for backing up and collaborating on codebases, and rolling back to earlier versions if required.
> 
> Version control tools meet the above needs. [Git](https://git-scm.com/) is an example of a version control tool, and [GitHub](https://github.com/) is a website + infrastructure that provides a Git server plus a number of really useful tools for working with git repositories individually or in teams, such as reporting issues with the code, reviewing tools, project management features such as assigning tasks and task statuses, and more.

Reviewer: - We can drop "Version control tools meet the above needs."
- Some long sentences that can be split up
- Added a few more project management features :)
```suggestion
Version control tools (often called **Version Control Systems** or **VCS**) are an essential part of modern workflows–for backing up code, collaborating on codebases, and rolling back to earlier versions if required.

[Git](https://git-scm.com/) is an example of a version control tool. [GitHub](https://github.com/), on the other hand, is a website and infrastructure that provides a Git server plus a number of useful tools for working with Git repositories, both individually and in teams. GitHub lets you report issues with the code, review code collaboratively, and offers project management features such as triaging issues, assigning tasks, planning projects, and more.
```

This module takes you through the essentials of version control using Git and GitHub.

## Prerequisites

- A desktop computer with Git installed on it (see the [Git downloads page](https://git-scm.com/downloads)).
- A tool to use Git. Depending on how you like to work, you could use:

>   - A [Git GUI client](https://git-scm.com/downloads/guis/) (we'd recommend [GitHub Desktop](https://desktop.github.com/download/), [SourceTree](https://www.sourcetreeapp.com/) or [Git Kraken](https://www.gitkraken.com/))

Reviewer: ```suggestion
  - A [Git GUI client](https://git-scm.com/downloads/guis/) (we'd recommend [GitHub Desktop](https://desktop.github.com/download/), [SourceTree](https://www.sourcetreeapp.com/) or [Git Kraken](https://www.gitkraken.com/)).
```

  - A command line/terminal window (check out our [Command line crash course](/en-US/docs/Learn_web_development/Getting_started/Environment_setup/Command_line) for an introduction).
- A [GitHub account](https://github.com/signup). If you haven't already got one, sign up now using the provided link.

## Guides

- [About GitHub and Git (from GitHub)](https://docs.github.com/en/get-started/start-your-journey/about-github-and-git)
  - : This covers what Git and GitHub are, how they work together, and how to get started.
- [Hello, World (from GitHub)](https://docs.github.com/en/get-started/start-your-journey/hello-world)

>   - : This practical guide jumps right into using GitHub, learning the basics of Git such as creating repositories and branches, making commits, and opening and merging pull requests.

Reviewer: ```suggestion
  - : This practical guide jumps right into using GitHub, teaching the basics of Git such as creating repositories and branches, making commits, and opening and merging pull requests.
```

- [Using Git (from GitHub)](https://docs.github.com/en/get-started/using-git)
  - : The Git Handbook goes into a little more depth, explaining what a version control tool is, what a repository is, how the basic GitHub model works, Git commands and examples, and more.
- [Contributing to a project (from GitHub)](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project)
  - : Forking projects is essential when you want to contribute to someone else's code. This guide explains how.
- [About Pull Requests (from GitHub)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests)

>   - : A useful guide to managing pull requests, why is how your suggested code changes are delivered to people's repositories for consideration.

Reviewer: - You probably meant "which is how"
- How about this phrasing (to remove delivery bit for simplification and to emphasize changes will be merged if accepted):
```suggestion
  - : A useful guide to managing pull requests. These requests are how you propose code changes to someone else's repository so they can review and potentially merge them into the main codebase.
```

- [About issues (from GitHub)](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues)
  - : Issues are like a forum for your GitHub project, where people can ask questions and report problems, and you can manage updates (for example assigning people to fix issues, clarifying the issue, letting people know things are fixed). This article tells you what you need to know about issues.

> [!NOTE]

> > As you get deeper into Git, you'll start to realize that it is easy to go wrong when you start using more complicated commands. Don't worry, even professional web developers find Git confusing sometimes, and often solve problems by searching for solutions on the web, or consulting sites like [Flight rules for Git](https://github.com/k88hudson/git-flight-rules) and [Dangit, git!](https://dangitgit.com/)

Reviewer: ```suggestion
> As you get deeper into Git, you'll start to realize that it is easy to go wrong when you start using more complicated commands. Don't worry, even professional web developers find Git confusing sometimes, and often solve problems by searching for solutions on the web, or consulting sites like [Flight rules for Git](https://github.com/k88hudson/git-flight-rules) and [Dangit, git!](https://dangitgit.com/).
```

## See also

> - Other useful topics from GitHub:

Reviewer: this to be more clear?
```suggestion
- Other useful topics covered in GitHub Docs include:
```

  - [Understanding the GitHub flow](https://docs.github.com/en/get-started/using-github/github-flow)
  - [Addressing merge conflicts](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts)
  - [Ignoring files with .gitignore](https://docs.github.com/en/get-started/git-basics/ignoring-files)
  - [Authentication to GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github)
- [Git command list](https://git-scm.com/docs)
- [Mastering markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) (the text format you write in on PR, issue comments, and `.md` files).

> - [Getting Started with GitHub Pages](https://docs.github.com/en/pages/quickstart) (how to publish demos and websites on GitHub).

Reviewer: These two are also from GitHub. Do you want to combine them with the "Other useful topics from GitHub" list?
